api/watcher/watcher.go

```diff
@@ -83,7 +84,10 @@ func (w *commonWatcher) commonLoop() {
 		defer wg.Done()
 		<-w.tomb.Dying()
 		if err := w.call("Stop", nil); err != nil {
-			logger.Errorf("error trying to stop watcher: %v", err)
+			// Don't log an error if a watcher is stopped due to an agent restart.
+			if err.Error() != worker.ErrRestartAgent.Error() {
```

Comment:
Because the error here is a comes from across the wire and just contains a string message